lib/Echidna/Processor.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Echidna.Processor where
+
+import Control.Arrow              (second)
+import Control.Lens
+import Control.Monad.IO.Class     (MonadIO(..))
+import Control.Exception          (Exception)
+import Control.Monad.Catch        (MonadThrow(..))
+import Data.Aeson                 (decode, Value(..))
+import System.Directory           (findExecutable)
+import System.Process             (StdStream(..), readCreateProcessWithExitCode, proc, std_err)
+import System.Exit                (ExitCode(..))
+import GHC.Word                   (Word32)
+
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import qualified Data.HashMap.Strict as M
+import qualified Data.Text           as T
+
+import Echidna.ABI (hashSig)
+
+-- | Things that can go wrong trying to run a processor. Read the 'Show'
+-- instance for more detailed explanations.
+data ProcException = ProcessorFailure String String
+                   | ProcessorNotFound String
+
+instance Show ProcException where
+  show = \case
+    ProcessorFailure p e     -> "Error running " ++ p ++ ": " ++ e
+    ProcessorNotFound p      -> "Cannot find processor " ++ p
+
+instance Exception ProcException
+
+filterResults :: Maybe String -> [(T.Text, [T.Text])] -> [Word32] 
+filterResults (Just c) rs = case lookup (T.pack c) rs of
```

Comment:
put the `case` on a newline, indent the cases to where `lookup` ends up being or two spaces

Reply:
Fixed.